### ISSUE TITLE
Socket callbacks require Sync auto-trait

### ISSUE THREAD

**Issue by fussybeaver** [abdolence/slack-morphism-rust#53]
Thanks for adding socket support recently. I gave it a try, though quite quickly ran into an issue - the callbacks on `SlackSocketModeListenerCallbacks` require the Sync auto-trait, which is not implemented on hyper's response Futures. So, you cannot simply use the `client` in a callback function to do arbitrary API requests:

```
async fn slack_command_events_fn(
    event: SlackCommandEvent,
    client: Arc<SlackHyperClient>,
    _states: Arc<RwLock<SlackClientEventsUserStateStorage>>,
) -> Result<SlackCommandEventResponse, Box<dyn std::error::Error + Send + Sync>> {
    let session = client.open_session(&...);
    let res = session
        .conversations_create(&SlackApiConversationsCreateRequest {
            name: "my_fancy_new_channel",
            is_private: Some(false),
            user_ds: None,
        })
        .await?;
    }
    // ...
}

#[tokio::main]
async fn main() -> Result<(), Box<dyn std::error::Error + Send + Sync>> {
    // ...    
    let socket_mode_callbacks = SlackSocketModeListenerCallbacks::new()
        .with_command_events(slack_command_events_fn);
    // ...
}

error[E0277]: `dyn Future<Output = Result<slack_morphism::api::SlackApiConversationsCreateResponse, Box<dyn std::error::Error + Send + Sync>>> + Send` cannot be shared between threads safely
   --> src/main.rs:140:10
    |
140 |         .with_command_events(slack_command_events_fn)
    |          ^^^^^^^^^^^^^^^^^^^ `dyn Future<Output = Result<slack_morphism::api::SlackApiConversationsCreateResponse, Box<dyn std::error::Error + Send + Sync>>> + Send` cannot be shared between threads safely
    |
    = help: the trait `Sync` is not implemented for `dyn Future<Output = Result<slack_morphism::api::SlackApiConversationsCreateResponse, Box<dyn std::error::Error + Send + Sync>>> + Send`
```

Let me know if I'm doing something wrong here...

**Comment by abdolence** [comment]
Interesting, thanks for testing it. It is a new thing, so might have some issues like this. Let me check your example.

**Comment by abdolence** [comment]
I think I fixed it with:
https://github.com/abdolence/slack-morphism-rust/commit/99749bb9c3434259c5071a660ccb11587b9eb426

And to make sure it won't happen again modified examples to have at least one call there.

**Comment by abdolence** [comment]
Fixed in v0.12.0. Please let me know if it works for you as well. The `Sync` part was copied from another place, and it wasn't required.

**Comment by fussybeaver** [comment]
Thanks for the quick fix!
